File: CNL/no.7 say hello(TCP)/TCPServer.java
//Write a program using TCP socket for wired network for following 
//Say Hello to Each other


import java.io.*;
import java.net.*;

public class TCPServer {
    public static void main(String[] args) {
        int port = 12345;

        try (ServerSocket serverSocket = new ServerSocket(port)) {
            System.out.println("Server is listening on port " + port + "...");
            Socket clientSocket = serverSocket.accept();
            System.out.println("Client connected.");

            PrintWriter out = new PrintWriter(clientSocket.getOutputStream(), true);
            BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));

            out.println("Server: Say Hello to Each Other");

            String message;
            while ((message = in.readLine()) != null) {
                System.out.println("Client: " + message);
                out.println("Server: Say Hello to Each Other");
            }

            clientSocket.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: CNL/no.7 say hello(TCP)/TCPServer.py
import socket

def tcp_server():
    # Create a TCP socket
    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    # Bind the socket to a specific address and port
    server_address = ('localhost', 12345)
    server_socket.bind(server_address)

    # Listen for incoming connections
    server_socket.listen(1)

    print("TCP Server is waiting for incoming connections...")

    while True:
        client_socket, client_address = server_socket.accept()  # Accept a connection from a client
        print(f"Connected to {client_address}")

        with open('received_file.txt', 'wb') as file:
            while True:
                data = client_socket.recv(1024)  # Receive data from the client
                if not data:
                    break
                file.write(data)  # Write received data to a file

        client_socket.close()
        print(f"Connection with {client_address} closed")

if __name__ == "__main__":
    tcp_server()
